### PR TITLE
Spotaut 16394 support bdm

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Added support for BDM

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-16394

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,6 +208,8 @@ No modules.
 | <a name="input_utilize_reserved_instances"></a> [utilize\_reserved\_instances](#input\_utilize\_reserved\_instances) | If there are any vacant Reserved Instances, launch On-Demand to consume them | `bool` | `true` | no |
 | <a name="input_whitelist"></a> [whitelist](#input\_whitelist) | List of instance types allowed in the Ocean cluster (`whitelist` and `blacklist` are mutually exclusive) | `list(string)` | `null` | no |
 | <a name="input_worker_instance_profile_arn"></a> [worker\_instance\_profile\_arn](#input\_worker\_instance\_profile\_arn) | The instance profile iam role. | `string` | n/a | yes |
+| <a name="input_block_device_mappings"></a> [block\_device\_mappings](#input\_block\_device\_mappings) | block\_device\_mapping object | <pre>list(object({<br>  device_name               = string<br>  delete_on_termination              = bool<br>  encrypted              = bool<br>  kms_key_id              = string<br>  snapshot_id              = string<br>  volume_type              = string<br>  iops                 = number<br>  volume_size              = number<br>  throughput              = number<br>  }))</pre> | `[]` | no |
```
Comment (chandra1-n) [inline README.md:211]: please correct the alignment

File: README.md [hunk 2]
```diff
@@ -192,6 +208,8 @@ No modules.
...
+| <a name="input_dynamic_volume_size"></a> [dynamic\_volume\_size](#input\_dynamic\_volume\_size) | dynamic\_volume\_size object | <pre>object({<br>base_size   = number<br>size_per_resource_unit = number<br>resource = string<br>    })</pre> | `null` | no |
```
Comment (chandra1-n) [inline README.md:212]: please correct the alignment

File: ocean.tf [hunk 3]
```diff
@@ -164,6 +164,31 @@ resource "spotinst_ocean_aws" "ocean" {
       }
     }
   }
+  ## Block Device Mappings ##
+  dynamic "block_device_mappings" {
+    for_each = var.block_device_mappings
```
Comment (chandra1-n) [inline ocean.tf:169]: add the below logic
!= null ? [var.block_device_mappings] : []

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: README.md (hunk 1 above)
Comment (sharadkesarwani) [reply README.md]: resolved alignment

File: README.md (hunk 2 above)
Comment (sharadkesarwani) [reply README.md]: resolved alignment

File: ocean.tf (hunk 3 above)
Comment (sharadkesarwani) [reply ocean.tf:169]: We will get error message, if we use above suggested logic.

Error: Unsupported attribute
│ 
│   on ../../ocean.tf line 171, in resource "spotinst_ocean_aws" "ocean":
│  171:       device_name = block_device_mappings.value.device_name
│     ├────────────────
│     │ block_device_mappings.value is list of object with 1 element
│ 
│ Can't access attributes on a list of objects. Did you mean to access attribute "device_name" for a specific element of the list, or across all elements of the list?

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**